baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LambdaMethodReference.java

```diff
@@ -91,49 +89,97 @@ public Description matchLambdaExpression(LambdaExpressionTree tree, VisitorState
 
     private Description checkMethodInvocation(
             MethodInvocationTree methodInvocation, LambdaExpressionTree root, VisitorState state) {
-        if (!methodInvocation.getArguments().isEmpty()
-                || !methodInvocation.getTypeArguments().isEmpty()) {
+        Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(methodInvocation);
+        if (methodSymbol == null || !methodInvocation.getTypeArguments().isEmpty()) {
             return Description.NO_MATCH;
         }
-        Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(methodInvocation);
-        if (methodSymbol == null || shouldIgnore(methodSymbol, root, methodInvocation)) {
+
+        if (methodInvocation.getArguments().isEmpty()
+                && (root.getParameters().size() == 1 || root.getParameters().isEmpty())) {
+            return convertSuppliersAndVariableInstanceMethods(methodSymbol, methodInvocation, root, state);
+        }
+
+        if (methodInvocation.getArguments().size() == root.getParameters().size()) {
+            return convertMethodInvocations(methodSymbol, methodInvocation, root, state);
+        }
+
+        return Description.NO_MATCH;
+    }
+
+    private Description convertSuppliersAndVariableInstanceMethods(
```

Comment:
Can we remove the supplier case here in favor of the zero-length method invocation code path?